src/core_modules/capture-core/components/Pages/EnrollmentEditEvent/EnrollmentEditEventPage.component.js

```diff
@@ -47,60 +59,137 @@ const EnrollmentEditEventPagePain = ({
+                onResetOrgUnitId={() => resetOrgUnitId()}
+                isUserInteractionInProgress={isUserInteractionInProgress}
+            >
+                <Grid item xs={12} sm={6} md={4} lg={2}>
+                    <SingleLockedSelect
+                        ready={pageStatus !== pageStatuses.MISSING_DATA}
+                        onClear={() => resetTeiId('/')}
+                        options={[
+                            {
+                                label: teiDisplayName,
+                                value: 'alwaysPreselected',
+                            },
+                        ]}
+                        selectedValue="alwaysPreselected"
+                        title={trackedEntityName}
+                        isUserInteractionInProgress={isUserInteractionInProgress}
+                    />
+                </Grid>
+                <Grid item xs={12} sm={6} md={4} lg={2}>
+                    <SingleLockedSelect
+                        ready={pageStatus !== pageStatuses.MISSING_DATA}
+                        onClear={() => resetEnrollmentId('enrollment')}
+                        options={enrollmentsAsOptions}
+                        selectedValue={enrollmentId}
+                        title={i18n.t('Enrollment')}
+                        isUserInteractionInProgress={isUserInteractionInProgress}
+                    />
+                </Grid>
+                <Grid item xs={12} sm={6} md={4} lg={2}>
+                    <SingleLockedSelect
+                        ready={pageStatus !== pageStatuses.MISSING_DATA}
+                        onClear={() => resetStageId('enrollment')}
+                        options={[
+                            {
+                                label: programStage?.name || '',
+                                value: 'alwaysPreselected',
+                            },
+                        ]}
+                        selectedValue="alwaysPreselected"
+                        title={i18n.t('stage')}
+                        isUserInteractionInProgress={isUserInteractionInProgress}
                     />
-                )}
-                {!hideWidgets.indicator && (
-                    <WidgetIndicator
-                        emptyText={i18n.t('There are no indicators for this event')}
-                        indicators={widgetEffects.indicators}
+                </Grid>
+                <Grid item xs={12} sm={6} md={4} lg={2}>
+                    <SingleLockedSelect
+                        ready={pageStatus !== pageStatuses.MISSING_DATA}
+                        onClear={() => resetEventId('enrollment')}
+                        options={[
+                            {
+                                label: eventDate || '',
+                                value: 'alwaysPreselected',
+                            },
+                        ]}
+                        selectedValue="alwaysPreselected"
+                        title={i18n.t('date of visit')}
```

Comment:
(Sorry for repeating myself)

We have some metadata that we can use for this title. It can be found here: programStage.stageForm.getLabel('eventDate')

(and yes we need to refactor where it resides at some point)